.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,28 @@
+name: Lint Go Code
+
+on:
+  pull_request:
+    paths:
+      - '**/*.go'
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.19.0
+
+    - name: Install golangci-lint
+      run: go get -u github.com/golangci/golangci-lint/cmd/golangci-lint
```

Comment:
I see there is a [github action](https://github.com/golangci/golangci-lint-action) for that which does [many awesome things](https://golangci-lint.run/usage/install/#ci-installation):

> We recommend using [our GitHub Action](https://github.com/golangci/golangci-lint-action) for running golangci-lint in CI for GitHub projects. It's [fast and uses smart caching](https://github.com/golangci/golangci-lint-action#performance) inside and it can be much faster than the simple binary installation.
> 
> Also, the action creates GitHub annotations for found issues: you don't need to dig into build log to see found by golangci-lint issues

Perhaps we can use it instead?